assets/src/edit-story/components/canvas/canvasUploadDropTarget.js

```diff
@@ -48,20 +49,54 @@ const PageAreaCover = styled(PageArea)`
 `;
 
 function CanvasUploadDropTarget({ children }) {
-  const { uploadFile } = useUploader();
+  const {
+    actions: { uploadMedia },
+  } = useMedia();
   const insertElement = useInsertElement();
+  const {
+    actions: { updateElementById, deleteElementById },
+  } = useStory();
+
+  const onLocalFile = useCallback(
```

Comment:
In this particular case, it'd be a lot easier, I think, to have a single `useMemo` that returns all three callbacks. Then they can all share the same structures, etc. Or, alternatively, I'd not mind if we inlined it all in the `onDropHandler` - then we will also be cleaner on scoping.